### PR TITLE
Api of batch add mappings

### PR DESCRIPTION
**Pull request by Pengpengwanga**
Signed-off-by: Pengpeng Wang <pengpengw@vmware.com>

### CODE REVIEW

File: flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java [hunk 1]
```diff
@@ -591,7 +594,13 @@ public void updateHostNameIPMapping(@RequestBody AssetIPMapping mapping) {
          pageSize = FlowgateConstant.maxPageSize;
       }
       PageRequest pageRequest = new PageRequest(pageNumber - 1, pageSize);
-      return assetIPMappingRepository.findAll(pageRequest);
+      if(ip == null) {
+         return assetIPMappingRepository.findAll(pageRequest);
+      }else if(IPAddressUtil.isValidIp(ip)) {
+         return assetIPMappingRepository.findByIp(ip, pageRequest);
+      }
+      throw new WormholeRequestException(HttpStatus.INTERNAL_SERVER_ERROR, "Invalid ip address",
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java:602]: add the ip address in the return message.

File: flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java [hunk 2]
```diff
@@ -600,6 +609,15 @@ public void deleteHostNameByID(@PathVariable("id") String id) {
       assetIPMappingRepository.delete(id);
    }
 
+   @RequestMapping(value = "/mapping/hostnameip/batch",method = RequestMethod.POST)
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java:612]: file

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 3]
```diff
@@ -53,6 +60,10 @@
    private AssetRepository assetRepository;
    @Autowired
    private AssetRealtimeDataRepository realtimeDataRepository;
+   private static final int CONTENT_LENGTH_PER_LINE = 2;
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:63]: let's check whether needed.

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 4]
```diff
@@ -210,6 +221,49 @@ public boolean isAssetNameValidate(String assetName) {
...
+   public List<AssetIPMapping> batchCreateMappingFromFile(MultipartFile multipartFile)
+         throws IOException {
+      List<AssetIPMapping> failureMappings = new ArrayList<AssetIPMapping>();
+      InputStream inputStream = multipartFile.getInputStream();
+      try (InputStreamReader inputStreamReader = new InputStreamReader(inputStream, "UTF-8");
+            BufferedReader br = new BufferedReader(inputStreamReader);) {
+         String assetIPMappingString = null;
+         while ((assetIPMappingString = br.readLine()) != null) {
+            AssetIPMapping mapping = null;
+            if(assetIPMappingString.contains(TAB)) {
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:233]: Let's refactor this.

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 5]
```diff
@@ -210,6 +221,49 @@ public boolean isAssetNameValidate(String assetName) {
...
+               assetIPMappingRepository.save(mapping);
+            }else {
+               failureMappings.add(mapping);
+            }
+         }
+      }
+      return failureMappings;
+   }
+
+   public static AssetIPMapping generateAssetIPMapping(String contentsArray[]) {
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:251]: parseAssetIPMapingByString()

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 6]
```diff
@@ -210,6 +221,49 @@ public boolean isAssetNameValidate(String assetName) {
...
+      AssetIPMapping mapping = new AssetIPMapping();
+      for(String content : contentsArray) {
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:253]: for 

!StringUtils.isempty(x){
if(mapping.ip is empty
}

Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:244]: Done

File: flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java (hunk 1 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java]: Done

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java (hunk 3 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java]: Removed

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java (hunk 5 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java]: Done

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java (hunk 4 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java]: Done

File: flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java (hunk 2 above)
Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java]: Done

**Comment by Pengpengwanga** [comment]
Closed #398

**Pull request merged**